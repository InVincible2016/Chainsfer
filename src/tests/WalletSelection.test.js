import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import update from 'immutability-helper'

import WalletSelection from '../components/WalletSelectionComponent'

const mockWallet = {
  connected: true,
  crypto: {
    ethereum:
    {
      0: {
        address: 'e7b7baf2b41c61f9f376ae3a5bc2cd78709995a7',
        balance: '10000000000000000',
        id: '0737b8f5-7f3e-46f8-8489-a60a8b9d3158',
        version: 3
      } },
    dai: {
      0: {
        address: 'e7b7baf2b41c61f9f376ae3a5bc2cd78709995a7',
        balance: '0',
        id: '0737b8f5-7f3e-46f8-8489-a60a8b9d3158',
        version: 3
      } },
    bitcoin: {
      0: {
        address: 'mtHrY25GUMvSsj9s72BQjPbK5iDvJykGFh',
        addresses: [],
        balance: '0',
        ciphertext: '6PYSS2TUay4BfyA3VBzidSmz5GqSbfe2cUoDobWomkL7acsb1AJbGRU6mK'
      }
    }
  }
}

const emptyWallet = {
  connected: false,
  crypto: {}
}

describe('WalletSelection', () => {
  let initialProps = {
    actionsPending: {
      checkWalletConnection: false,
      syncAccountInfo: false,
      updateBtcAccountInfo: false,
      checkCloudWalletConnection: false
    },
    cryptoSelectionPrefilled: undefined,
    cryptoType: null,
    error: '',
    goToStep: () => {},
    handleNext: () => {},
    onCryptoSelected: () => {},
    onWalletSelected: () => {},
    wallet: undefined,
    walletType: null
  }
  let createRenderTree = (props) => {
    return renderer
      .create(
        <MemoryRouter>
          <WalletSelection {...props} />
        </MemoryRouter>
      )
  }

  it('initial render', () => {
    const props = initialProps
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // drive wallet
  it('select drive', () => {
    const props = update(initialProps, { walletType: { $set: 'drive' } })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select drive and ETH', () => {
    const props = update(initialProps, {
      walletType: { $set: 'drive' },
      cryptoType: { $set: 'ethereum' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select drive and DAI', () => {
    const props = update(initialProps, {
      walletType: { $set: 'drive' },
      cryptoType: { $set: 'dai' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select drive and BTC', () => {
    const props = update(initialProps, {
      walletType: { $set: 'drive' },
      cryptoType: { $set: 'bitcoin' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // Metamask
  it('select metamask', () => {
    const props = update(initialProps, { walletType: { $set: 'metamask' } })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select metamask and ETH', () => {
    const props = update(initialProps, {
      walletType: { $set: 'metamask' },
      cryptoType: { $set: 'ethereum' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select metamask and DAI', () => {
    const props = update(initialProps, {
      walletType: { $set: 'metamask' },
      cryptoType: { $set: 'dai' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // ledger
  it('select ledger', () => {
    const props = update(initialProps, { walletType: { $set: 'ledger' } })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select ledger and ETH', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'ethereum' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select ledger and DAI', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'dai' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('select ledger and BTC', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'bitcoin' },
      wallet: { $set: mockWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // actionPending
  it('checkWalletConnection', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'bitcoin' },
      actionsPending: { checkWalletConnection: { $set: true } },
      wallet: { $set: emptyWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('syncAccountInfo', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'bitcoin' },
      actionsPending: { syncAccountInfo: { $set: true } },
      wallet: { $set: emptyWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('updateBtcAccountInfo', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'bitcoin' },
      actionsPending: { updateBtcAccountInfo: { $set: true } },
      wallet: { $set: emptyWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('checkCloudWalletConnection', () => {
    const props = update(initialProps, {
      walletType: { $set: 'ledger' },
      cryptoType: { $set: 'bitcoin' },
      actionsPending: { checkCloudWalletConnection: { $set: true } },
      wallet: { $set: emptyWallet }
    })
    const tree = createRenderTree(props).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
