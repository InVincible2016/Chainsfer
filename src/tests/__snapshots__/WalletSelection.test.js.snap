// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletSelectionComponent rendering initial render without error 1`] = `
<div
  class="MuiGrid-container-13 MuiGrid-spacing-xs-24-37 MuiGrid-direction-xs-column-16 MuiGrid-justify-xs-center-30"
>
  <div
    class="MuiGrid-item-14"
  >
    <div
      class="MuiGrid-container-13"
    >
      <div
        class="MuiGrid-item-14"
      >
        <h6
          class="MuiTypography-root-110 MuiTypography-h6-127 MuiTypography-alignLeft-132"
        >
          Choose your wallet
        </h6>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-14"
  >
    <div
      class="MuiGrid-container-13 MuiGrid-align-items-xs-center-21 MuiGrid-justify-xs-center-30"
    >
      <div
        class="MuiGrid-item-14"
      >
        <div
          class="MuiGrid-container-13 MuiGrid-direction-xs-column-16 MuiGrid-align-items-xs-center-21 SquareButtonComponent-btn-146"
          jutify="center"
        >
          <div
            class="MuiGrid-item-14 SquareButtonComponent-imgContainer-156"
          >
            <img
              alt="wallet-logo"
              class="SquareButtonComponent-btnLogo-149"
              src="drive-wallet-button.png"
            />
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnTitle-150"
            >
              Drive
            </p>
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnDesc-151"
            >
              Use Drive Wallet
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-14"
      >
        <div
          class="MuiGrid-container-13 MuiGrid-direction-xs-column-16 MuiGrid-align-items-xs-center-21 SquareButtonComponent-btn-146"
          jutify="center"
        >
          <div
            class="MuiGrid-item-14 SquareButtonComponent-imgContainer-156"
          >
            <img
              alt="wallet-logo"
              class="SquareButtonComponent-btnLogo-149"
              src="metamask-button.png"
            />
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnTitle-150"
            >
              Metamask
            </p>
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnDesc-151"
            >
              MetaMask Extension
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-14"
      >
        <div
          class="MuiGrid-container-13 MuiGrid-direction-xs-column-16 MuiGrid-align-items-xs-center-21 SquareButtonComponent-btn-146"
          jutify="center"
        >
          <div
            class="MuiGrid-item-14 SquareButtonComponent-imgContainer-156"
          >
            <img
              alt="wallet-logo"
              class="SquareButtonComponent-btnLogo-149"
              src="ledger-button.png"
            />
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnTitle-150"
            >
              Ledger
            </p>
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnDesc-151"
            >
              Ledger Hardware Wallet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-14"
  >
    <div
      class="MuiGrid-container-13 MuiGrid-spacing-xs-24-37 MuiGrid-justify-xs-center-30"
    >
      <div
        class="MuiGrid-item-14"
      >
        <button
          class="MuiButtonBase-root-183 MuiButton-root-157 MuiButton-text-159 MuiButton-textPrimary-160 MuiButton-flat-162 MuiButton-flatPrimary-163"
          tabindex="0"
          to="/"
          type="button"
        >
          <span
            class="MuiButton-label-158"
          >
            Cancel Transfer
          </span>
          <span
            class="MuiTouchRipple-root-186"
          />
        </button>
      </div>
      <div
        class="MuiGrid-item-14"
      >
        <button
          class="MuiButtonBase-root-183 MuiButtonBase-disabled-184 MuiButton-root-157 MuiButton-contained-168 MuiButton-containedPrimary-169 MuiButton-raised-171 MuiButton-raisedPrimary-172 MuiButton-sizeLarge-181 MuiButton-disabled-177 MuiButton-fullWidth-182"
          disabled=""
          id="continue"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label-158"
          >
            Continue
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`WalletSelectionComponent rendering select ledger and DAI 1`] = `
<div
  class="MuiGrid-container-13 MuiGrid-spacing-xs-24-37 MuiGrid-direction-xs-column-16 MuiGrid-justify-xs-center-30"
>
  <div
    class="MuiGrid-item-14"
  >
    <div
      class="MuiGrid-container-13"
    >
      <div
        class="MuiGrid-item-14"
      >
        <h6
          class="MuiTypography-root-110 MuiTypography-h6-127 MuiTypography-alignLeft-132"
        >
          Choose your wallet
        </h6>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-14"
  >
    <div
      class="MuiGrid-container-13 MuiGrid-align-items-xs-center-21 MuiGrid-justify-xs-center-30"
    >
      <div
        class="MuiGrid-item-14"
      >
        <div
          class="MuiGrid-container-13 MuiGrid-direction-xs-column-16 MuiGrid-align-items-xs-center-21 SquareButtonComponent-btn-146"
          jutify="center"
        >
          <div
            class="MuiGrid-item-14 SquareButtonComponent-imgContainer-156"
          >
            <img
              alt="wallet-logo"
              class="SquareButtonComponent-btnLogo-149"
              src="drive-wallet-button.png"
            />
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnTitle-150"
            >
              Drive
            </p>
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnDesc-151"
            >
              Use Drive Wallet
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-14"
      >
        <div
          class="MuiGrid-container-13 MuiGrid-direction-xs-column-16 MuiGrid-align-items-xs-center-21 SquareButtonComponent-btn-146"
          jutify="center"
        >
          <div
            class="MuiGrid-item-14 SquareButtonComponent-imgContainer-156"
          >
            <img
              alt="wallet-logo"
              class="SquareButtonComponent-btnLogo-149"
              src="metamask-button.png"
            />
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnTitle-150"
            >
              Metamask
            </p>
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnDesc-151"
            >
              MetaMask Extension
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-14"
      >
        <div
          class="MuiGrid-container-13 MuiGrid-direction-xs-column-16 MuiGrid-align-items-xs-center-21 SquareButtonComponent-btnSelected-147"
          jutify="center"
        >
          <div
            class="MuiGrid-item-14 SquareButtonComponent-imgContainer-156"
          >
            <img
              alt="wallet-logo"
              class="SquareButtonComponent-btnLogo-149"
              src="ledger-button.png"
            />
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnTitleSelected-154"
            >
              Ledger
            </p>
          </div>
          <div
            class="MuiGrid-item-14"
          >
            <p
              class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-alignCenter-133 SquareButtonComponent-btnDescSelected-155"
            >
              Ledger Hardware Wallet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-14"
  >
    <h6
      class="MuiTypography-root-110 MuiTypography-h6-127 MuiTypography-alignLeft-132"
    >
      Choose the coin
    </h6>
  </div>
  <div
    class="MuiGrid-item-14"
  >
    <ul
      class="MuiList-root-193 MuiList-padding-194 WalletSelectionComponent-cryptoList-2"
    >
      <div>
        <hr
          class="MuiDivider-root-197"
        />
        <div
          class="MuiButtonBase-root-183 MuiListItem-root-202 MuiListItem-default-205 MuiListItem-gutters-210 MuiListItem-button-211"
          id="ethereum"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiGrid-container-13 MuiGrid-align-items-xs-center-21 MuiGrid-justify-xs-space-between-32"
          >
            <span
              class="MuiButtonBase-root-183 MuiIconButton-root-223 MuiPrivateSwitchBase-root-219 MuiRadio-root-214 MuiRadio-colorPrimary-217"
            >
              <span
                class="MuiIconButton-label-228"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-229"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  class="MuiPrivateSwitchBase-input-222"
                  id="ethereum"
                  tabindex="-1"
                  type="radio"
                  value=""
                />
              </span>
            </span>
            <div
              class="MuiListItemText-root-238"
            >
              <span
                class="MuiTypography-root-110 MuiTypography-subheading-117 MuiListItemText-primary-241"
              >
                ETH
              </span>
              <p
                class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-colorTextSecondary-143 MuiListItemText-secondary-242"
              >
                Ethereum
              </p>
            </div>
          </div>
          <span
            class="MuiTouchRipple-root-186"
          />
        </div>
      </div>
      <div>
        <hr
          class="MuiDivider-root-197"
        />
        <div
          class="MuiButtonBase-root-183 MuiListItem-root-202 MuiListItem-default-205 MuiListItem-gutters-210 MuiListItem-button-211"
          id="dai"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiGrid-container-13 MuiGrid-align-items-xs-center-21 MuiGrid-justify-xs-space-between-32"
          >
            <span
              class="MuiButtonBase-root-183 MuiIconButton-root-223 MuiPrivateSwitchBase-root-219 MuiRadio-root-214 MuiRadio-colorPrimary-217"
            >
              <span
                class="MuiIconButton-label-228"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-229"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  class="MuiPrivateSwitchBase-input-222"
                  id="dai"
                  tabindex="-1"
                  type="radio"
                  value=""
                />
              </span>
            </span>
            <div
              class="MuiListItemText-root-238"
            >
              <span
                class="MuiTypography-root-110 MuiTypography-subheading-117 MuiListItemText-primary-241"
              >
                DAI
              </span>
              <p
                class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-colorTextSecondary-143 MuiListItemText-secondary-242"
              >
                Dai
              </p>
            </div>
          </div>
          <span
            class="MuiTouchRipple-root-186"
          />
        </div>
      </div>
      <div>
        <hr
          class="MuiDivider-root-197"
        />
        <div
          class="MuiButtonBase-root-183 MuiListItem-root-202 MuiListItem-default-205 MuiListItem-gutters-210 MuiListItem-button-211"
          id="bitcoin"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiGrid-container-13 MuiGrid-align-items-xs-center-21 MuiGrid-justify-xs-space-between-32"
          >
            <span
              class="MuiButtonBase-root-183 MuiIconButton-root-223 MuiPrivateSwitchBase-root-219 MuiRadio-root-214 MuiRadio-colorPrimary-217 MuiPrivateSwitchBase-checked-220 MuiRadio-checked-215"
            >
              <span
                class="MuiIconButton-label-228"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-229"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked=""
                  class="MuiPrivateSwitchBase-input-222"
                  id="bitcoin"
                  tabindex="-1"
                  type="radio"
                  value=""
                />
              </span>
            </span>
            <div
              class="MuiListItemText-root-238"
            >
              <span
                class="MuiTypography-root-110 MuiTypography-subheading-117 MuiListItemText-primary-241"
              >
                BTC
              </span>
              <p
                class="MuiTypography-root-110 MuiTypography-body1-119 MuiTypography-colorTextSecondary-143 MuiListItemText-secondary-242"
              >
                Bitcoin
              </p>
            </div>
          </div>
          <span
            class="MuiTouchRipple-root-186"
          />
        </div>
      </div>
      <hr
        class="MuiDivider-root-197"
      />
    </ul>
  </div>
  <div
    class="MuiGrid-item-14"
  />
  <div
    class="MuiGrid-item-14"
  >
    <div
      class="MuiGrid-container-13 MuiGrid-spacing-xs-24-37 MuiGrid-justify-xs-center-30"
    >
      <div
        class="MuiGrid-item-14"
      >
        <button
          class="MuiButtonBase-root-183 MuiButton-root-157 MuiButton-text-159 MuiButton-textPrimary-160 MuiButton-flat-162 MuiButton-flatPrimary-163"
          tabindex="0"
          to="/"
          type="button"
        >
          <span
            class="MuiButton-label-158"
          >
            Cancel Transfer
          </span>
          <span
            class="MuiTouchRipple-root-186"
          />
        </button>
      </div>
      <div
        class="MuiGrid-item-14"
      >
        <button
          class="MuiButtonBase-root-183 MuiButtonBase-disabled-184 MuiButton-root-157 MuiButton-contained-168 MuiButton-containedPrimary-169 MuiButton-raised-171 MuiButton-raisedPrimary-172 MuiButton-sizeLarge-181 MuiButton-disabled-177 MuiButton-fullWidth-182"
          disabled=""
          id="continue"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label-158"
          >
            Continue
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
